controllers/hcloudmachine_controller_test.go

```diff
@@ -808,6 +809,21 @@ var _ = Describe("HCloudMachine validation", func() {
 		hcloudMachine.Spec.ImageName = ""
 		Expect(testEnv.Create(ctx, hcloudMachine)).ToNot(Succeed())
 	})
+	It("should allow valid HCloudMachine creation", func() {
+		Expect(testEnv.Create(ctx, hcloudMachine)).To(Succeed())
+	})
+	It("should prevent updating immutable fields", func() {
+		Expect(testEnv.Create(ctx, hcloudMachine)).To(Succeed())
+
+		Eventually(func() error {
+			key := client.ObjectKey{Namespace: testNs.Name, Name: hcloudMachine.Name}
+			return testEnv.Client.Get(ctx, key, hcloudMachine)
+		}, timeout, time.Second).Should(BeNil())
```

Comment:
don't use "time.Second" but the variable. Is it called "interval"? defined together with "timeout"

Reply:
it was not there in the controlers_suite_test file , will add it there and use that instead